server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -648,8 +648,10 @@ public void lock(List<Pool> poolsToLock) {
             "FROM Pool AS Pool " +
             "  INNER JOIN Pool.product AS Product " +
             "  INNER JOIN Product.attributes AS Attribute " +
+            "  LEFT JOIN Pool.entitlements AS Entitlement " +
```

Comment:
This looks to have been addressed already, but for clarity, the DISTINCT issue applies to both HQL and Criteria, but only in the case where we're using ScrollableResults.